libs/ui/src/search_select.tsx

```diff
@@ -0,0 +1,342 @@
+import React, { useRef, useState } from 'react';
+import styled from 'styled-components';
+import { assert, assertDefined } from '@votingworks/basics';
+import { Icons } from './icons';
+
+function noop() {
+  // do nothing
+}
+
+export interface SelectOption<T extends string = string> {
+  value: T;
+  label: string;
+}
+
+const ComponentContainer = styled.div`
+  width: 100%;
+  position: relative;
+`;
+
+const SelectBar = styled.div`
+  width: 100%;
+  min-height: 2.2rem;
+  display: flex;
+  justify-content: space-between;
+  align-items: center;
+  border: ${(p) => p.theme.sizes.bordersRem.medium}rem solid
+    ${(p) => p.theme.colors.foreground};
+  border-radius: 0.25rem;
+  background: ${(p) => p.theme.colors.background};
+`;
+
+const ValuesContainer = styled.div<{
+  isMulti: boolean;
+}>`
+  display: ${(p) => (p.isMulti ? 'flex' : 'grid')};
+  flex: 1 1 0%;
+  flex-wrap: wrap;
+  align-items: center;
+  gap: 0.1rem;
+  margin: 0.2rem 0.3rem;
+  border-radius: 0.25rem;
+  overflow: hidden;
+`;
+
+const INPUT_CONTAINER_GRID_AREA = '1 / 1 / 2 / 3';
+const INPUT_GRID_AREA = '1 / 2 / auto / auto';
```

Comment:
The problem of a multi-select is that you need an `input` component that's going to resize as you type, going to the next line if necessary (forgot to demo this in video but it's neat!). But `input` components normally don't resize.

Here's how `react-select` handles this while not totally running afoul of semantic HMTL. Basically, we create a grid and snap multiple elements to the _same_ grid-area, which allows the `input`s to resize according to another element, even though `input`s normally do not resize.

`INPUT_CONTAINER_GRID_AREA` is for, in the single select case, colocating the current value and the input.

`INPUT_GRID_AREA` is for, in the multi-select case`, resizing the `input` according to a computed `::after` element of the `InputContainer` that is based on the current search string's length.